analyzers/src/SonarAnalyzer.CSharp/Rules/PureAttributeOnVoidMethod.cs

```diff
@@ -50,7 +51,8 @@ protected override void Initialize(SonarAnalysisContext context)
                 SyntaxKindEx.LocalFunctionStatement);
         }
 
-        private static bool IsPureAttribute(string attributeName) =>
-            attributeName.Equals("Pure") || attributeName.Equals("PureAttribute");
+        private static bool IsPureAttribute(AttributeSyntax attribute) =>
+            attribute.Name.GetIdentifier() is { } name
+            && (name.Identifier.ValueText == "Pure" || name.Identifier.ValueText == "PureAttribute");
```

Comment:
Don't we want to check here also the actual symbol? (Otherwise we will have false positives with user-defined attributes named `Pure`)

Reply:
`InvalidPureDataAttributeUsage` does that inside (see `GetPureAttribute` in the base class)

Reply:
I've added UT